### DISCUSSION TITLE
Imagemagick library causes core dump with php and imagick extension on command line php

### DISCUSSION THREAD

**Discussion by 73blazer** [ImageMagick/ImageMagick#2961] [Development]
When I run any command line php, like 'php -r "phpinfo();" I get a core dump if the extension imagick is loaded. Interactive web php/apache with all the imagick extensions work just fine. Only command line php fails. 

So, this is definitely something to do with openmp on AIX (my platform). It's not the extension, as if I simply swap out imagemagick base for one built without openmp (with or without re-compiling the extension or php), everything works. 
I've tried compiling this many different ways. 32 bit, 64bit, with gcc with native ibm xlc compiler, various compiler and link options. All produce the same results. --disable-openmp when building imagemagick and that build works fine. And not just compiling imagemagick, I've tried compiling apache/php/imagemagick and it's dependencies and the imagic extension in various modes/methods, nothing seems to change the coredump behavior. 
I've also tried the aix toolbox for linux pre-built imagemagick which is even worse, that breaks web php any imagick call will spin forever and hang up apache. 

I'm not convinced it's a bug in imagemagicck, but I've no clue on how to go about figuring out why it core dumps upon command line php calls. Any ideas on debugging or anything to try with the env??????

Env currently is AIX 7.2 OS, apache 2.4.46, imagemagick 7.0.10.47 (43,46,47 all produce same results) , imagick php extension 3.4.4, php 7.4.13 (12 and 11 are same results) all compiled 64bit currently from the native ibm xlc compiler (it performs just a bit better under xlc) . gcc will do the same core dump. .

**Comment by urban-warrior** [comment]
Try running php from gdb.  When it faults, type `bt` to look at the backtrace.  It should tell you where in the stack the fault occurs.  You can also try restart httpd or php-frm.  Either may have an older ImageMagick shared library in core that is causing problems.
